OpenRA.Mods.Common/Traits/Production.cs

```diff
@@ -108,7 +108,7 @@ public virtual void DoProduction(Actor self, ActorInfo producee, ExitInfo exitin
 
 		protected virtual Exit SelectExit(Actor self, ActorInfo producee, string productionType, Func<Exit, bool> p)
 		{
-			return self.RandomExitOrDefault(self.World, productionType, p);
+			return self.ClosestExitOrDefault(productionType, rp.Value.Location, p);
```

Comment:
This is a clear benefit for the naval yard and sub pen, but feels like a polish regression for the barracks in both RA and TD - the two doors are basically equivalent, but we now only ever use one at a time.

Can we please implement
```csharp
[Desc("Select a random exit instead of always using the closest to the rally point.")]
public bool RandomExit = false;
```

in `ProductionInfo` and then use it here to switch between `RandomExitOrDefault` vs `ClosestExitOrDefault`? Then enable this on `BARR` and `TENT` in RA and `PYLE` in TD.